tls/s2n_cipher_suites.c

```diff
@@ -1209,58 +1324,52 @@ static int s2n_set_cipher_as_server(struct s2n_connection *conn, uint8_t *wire,
     const struct s2n_security_policy *security_policy;
     POSIX_GUARD(s2n_connection_get_security_policy(conn, &security_policy));
 
-    /* s2n supports only server order */
-    for (int i = 0; i < security_policy->cipher_preferences->count; i++) {
-        const uint8_t *ours = security_policy->cipher_preferences->suites[i]->iana_value;
+    /* Check if both server and client signaled support for ChaCha20 boosting */
+    do {
```

Comment:
This needs to be moved out of the main s2n_set_cipher_as_server method. It's a minor use case, an exception rather than the rule, and it shouldn't take over the method.

The diff here is extremely hard to follow with your changes added, and it shouldn't need to be.

Reply:
Sure, let me pull it out into a function